de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandInvalidUpdater.java

```diff
@@ -356,23 +354,23 @@ private void populateC_InvoiceCandidate_InOutLine(
 		final List<I_M_InOutLine> inoutLines = inOutDAO.retrieveLinesForOrderLine(orderLine, I_M_InOutLine.class);
 		for (final I_M_InOutLine inOutLine : inoutLines)
 		{
-			if (invoiceCandDAO.existsInvoiceCandidateInOutLinesForInvoiceCandidate(ic, inOutLine))
+			// create a new PO or update the unique existing one
+			I_C_InvoiceCandidate_InOutLine iciol = invoiceCandDAO.retrieveInvoiceCandidateInOutLine(ic, inOutLine);
+			if (iciol == null)
```

Comment:
here is the fix, right?

Reply:
Yes: the problem was that existing `I_C_InvoiceCandidate_InOutLine` were not updated with new/updated catch weight data.